src/PrimaryGeneration.cpp

```diff
@@ -186,9 +200,188 @@ void PrimaryGeneration::GeneratePrimaries(G4Event* event)
+  G4bool override_vertex_position_ = ConfigManager::GetOverrideVertexPosition();
+  G4bool printParticleInfo_ = ConfigManager::GetPrintParticleInfo();
+  G4double vertex_x_ = ConfigManager::GetVertexX();
+  G4double vertex_y_ = ConfigManager::GetVertexY();
+  G4double vertex_z_ = ConfigManager::GetVertexZ();
+  G4String particleType = ConfigManager::GetParticleType();
+  G4ThreeVector momentumDirection_ = ConfigManager::GetMomentumDirection();
+
+  particleType.toLower();
+
+
+  // Get MCTruthManager
+  MCTruthManager * mc_truth_manager = MCTruthManager::Instance();
+
+  GENIEManager * genieManager = GENIEManager::Instance();
+
+  genieManager->Cd();
+
+  genieManager->LoadEvent(event);
+
+  G4ThreeVector vertex3d = G4ThreeVector (vertex_x_, vertex_y_, vertex_z_);
+
+  G4PrimaryVertex * vertex = new G4PrimaryVertex(vertex3d, 0);
+
+  for (int np = 0; np<genieManager->GetNParticles_(); np++){
+    if (genieManager->GetIdx_(np)<0)
+    {
+      G4cout<<"Skipping this particle PDGCode-> "<< genieManager->GetPDG_(np)<< " Idx-> "<<genieManager->GetIdx_(np)<<" Status= "<<genieManager->GetStatus_(np)<<G4endl;
+      continue;
+    }
+
+    // If the particle is a nucleus
+    G4ParticleDefinition *pdef=0;
+
+    if (genieManager->GetPDG_(np) > 1000000000 && genieManager->GetPDG_(np)<2000000000)
+    {
+      if (!pdef)
+      {
+        int const Z = (genieManager->GetPDG_(np) % 10000000) / 10000; // atomic number
+        int const A = (genieManager->GetPDG_(np) % 10000) / 10; // mass number
+        pdef = particle_table_->GetIonTable()->GetIon(Z, A, 0.);
+      }
+    } 
+    else
+    {
+      pdef = particle_table_->FindParticle(genieManager->GetPDG_(np));
+    }
+
+    if (pdef==0){
+      G4String str1="Couldnt find the particle table for " + std::to_string(genieManager->GetPDG_(np));
+      G4Exception("[PrimaryGeneration]","GENIEGeneratePrimaries",G4ExceptionSeverity::JustWarning,str1.c_str());
+      continue;
+    }
+
+    // Print out info about the GENIE particle
+    //G4cout << "..........oooooOOO000OOOooooo.........." << G4endl
+    //       << "GENIE Particle Info:" << G4endl
+    //       << "  PDG Code:       " << genieManager->GetPDG_(np) << G4endl
+    //       << "  Energy:         " << genieManager->GetE_(np) << G4endl
+    //       << "  Px:             " << genieManager->GetPx_(np) << G4endl
+    //       << "  Py:             " << genieManager->GetPy_(np) << G4endl
+    //       << "  Pz:             " << genieManager->GetPz_(np) << G4endl
+    //       << "G4ParticleDefinition Info:" << G4endl
+    //       << "  Mass:           " << pdef->GetPDGMass() << G4endl
+    //       << "  Charge:         " << pdef->GetPDGCharge() << G4endl
+    //       << "User defined Info:" << G4endl
+    //       << "  Vertex:         " << vertex3d << G4endl
+    //       << "..........oooooOOO000OOOooooo.........." << G4endl;
+
+
+    // filter the particles less than 1 eV
+    // if(genieManager->GetE_(np)<(1 *CLHEP::eV)) continue;
```

Comment:
Don't keep commented out code in the repository.  You could protect the printout with a verbosity flag.

Reply:
fixing in bugfix/v3.0.0_revisions branch